analyzers/src/SonarAnalyzer.CSharp/Rules/InsteadOfAny.cs

```diff
@@ -25,6 +25,17 @@ public sealed class InsteadOfAny : InsteadOfAnyBase<SyntaxKind, InvocationExpres
 {
     protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
+    protected override HashSet<SyntaxKind> ExitParentKinds => new()
+    {
+        SyntaxKind.MethodDeclaration,
+        SyntaxKind.ConstructorDeclaration,
+        SyntaxKind.DestructorDeclaration,
+        SyntaxKind.GetAccessorDeclaration,
+        SyntaxKind.SetAccessorDeclaration,
```

Comment:
nitpick: I guess to be explicit you could add `SyntaxKindEx.InitAccessorDeclaration`, but anyhow there might always be some edge cases and the algorithm will fallback on CompilationUnit so that's fine